lib/ansible/modules/system/user.py

```diff
@@ -1001,11 +1002,22 @@ def create_homedir(self, path):
                     shutil.copytree(skeleton, path, symlinks=True)
                 except OSError as e:
                     self.module.exit_json(failed=True, msg="%s" % to_native(e))
-        else:
-            try:
-                os.makedirs(path)
-            except OSError as e:
-                self.module.exit_json(failed=True, msg="%s" % to_native(e))
+            else:
+                try:
+                    os.makedirs(path)
+                except OSError as e:
+                    self.module.exit_json(failed=True, msg="%s" % to_native(e))
+            # get umask from /etc/login.defs and set correct home mode
+            if os.path.exists(self.LOGIN_DEFS):
+                for line in open(self.LOGIN_DEFS).readlines():
```

Comment:
Use a context manager with this read operation to ensure the file handler is closed as soon as it is no longer needed rather than relying on successful program termination.
```suggestion
                with open(self.LOGIN_DEFS, 'r') as f:
                    for line in f.readlines():
```

Reply:
Thanks, I have no idea why I didn't use it, because I remember doing it :D

Reply:
While I was fixing this, I updated all file operations to use `with` because it was all over the place.